### DISCUSSION TITLE
Albums Newest or Oldest Sort Order

### DISCUSSION THREAD

**Discussion by michaelcresswell** [photoprism/photoprism#3487] [Q&A]
I have been creating albums for older photos in my library. I have created my albums in a different order than they occurred (i.e. I created albums for events in 2022 before events in 2021, etc.).

When I sort albums 'Newest First' or 'Oldest First', their sort order matches newest or oldest based on when they were created in Photoprism.  I am referring to the highlighted options as seen on the demo site:
![image](https://github.com/photoprism/photoprism/assets/3800846/979d3fb6-5b2b-4367-b1f1-a6d8a0720829)

Manually changing any album datetime values in the database does not seem to affect the newest or oldest sort order.
- https://github.com/photoprism/photoprism/blob/develop/internal/entity/schema/mariadb.sql#L27-L29
- https://github.com/photoprism/photoprism/blob/develop/internal/entity/schema/mariadb.sql#L34-L35
- https://github.com/photoprism/photoprism/blob/develop/internal/entity/schema/mariadb.sql#L38

I believe this section of code is responsible for album sort order:
```
	case sortby.Moment, sortby.Newest:
		if f.Type == entity.AlbumManual || f.Type == entity.AlbumState {
			s = s.Order("albums.album_uid DESC")
		} else if f.Type == entity.AlbumMoment {
			s = s.Order("has_year, albums.album_year DESC, albums.album_month DESC, albums.album_day DESC, albums.album_title, albums.album_uid DESC")
		} else {
			s = s.Order("albums.album_year DESC, albums.album_month DESC, albums.album_day DESC, albums.album_title, albums.album_uid DESC")
		}
```
https://github.com/photoprism/photoprism/blob/develop/internal/search/albums.go#L87-L102

Is there a reason that album types like Moments consider an album year, month, and day ahead of the uid while others like user-defined manual albums sort based on uid alone?

For reference, I am self-hosting Build 230615-90a18f6e7 via Docker and do not believe this behavior introduced by any recent updates.

**Comment by lastzero** [comment]
Note that the year, month, and day are determined automatically based on the album contents (edit: for certain album types like Moments). This is why there isn't an input field. Any changes you perform in the database or code are your own responsibility and must take this into account. In a future version, we could make these fields be editable for manually created albums. However, I can see that this will lead to additional feature requests in that users want to specify ranges, repeated events like birthdays, etc. We don't have the resources to implement and release all this short term.
